node-manager/src/wallet.rs

```diff
@@ -0,0 +1,83 @@
+use std::str::FromStr;
+
+use bdk::blockchain::EsploraBlockchain;
+use bdk::keys::ExtendedKey;
+use bdk::template::DescriptorTemplateOut;
+use bdk::{SyncOptions, Wallet};
+use bip39::Mnemonic;
+use bitcoin::util::bip32::{ChildNumber, DerivationPath, ExtendedPrivKey};
+use bitcoin::Network;
+
+use crate::localstorage::MutinyBrowserStorage;
+
+#[derive(Debug)]
+pub struct MutinyWallet {
+    pub wallet: Wallet<MutinyBrowserStorage>,
+}
+
+impl MutinyWallet {
+    pub fn new(mnemonic: Mnemonic, network: Network) -> MutinyWallet {
+        let entropy = mnemonic.to_entropy();
+        let xprivkey = ExtendedPrivKey::new_master(network, &entropy).unwrap();
+        let xkey = ExtendedKey::from(xprivkey);
+        let account_number = 0;
+        let (receive_descriptor_template, change_descriptor_template) =
+            get_tr_descriptors_for_extended_key(xkey, network, account_number);
+
+        let wallet = Wallet::new(
+            receive_descriptor_template,
+            Some(change_descriptor_template),
+            network,
+            MutinyBrowserStorage::new(),
+        )
+        .expect("Error creating wallet");
+
+        MutinyWallet { wallet }
+    }
+
+    pub async fn sync(&self) -> Result<(), bdk::Error> {
+        let url = match self.wallet.network() {
+            Network::Bitcoin => Ok("https://blockstream.info/api"),
+            Network::Testnet => Ok("https://blockstream.info/testnet/api"),
+            Network::Signet => Ok("https://mempool.space/signet/api"),
+            Network::Regtest => Err(bdk::Error::Generic(
+                "No esplora client available for regtest".to_string(),
+            )),
+        }?;
+
+        let blockchain = EsploraBlockchain::new(url, 20);
+        self.wallet.sync(&blockchain, SyncOptions::default()).await
```

Comment:
this function is only async if the target is `wasm32`, for POC probably fine to ignore but is there something we can do to handle this for different builds?

Reply:
yeah we'd need an async executor like async-std or tokio if cfg not wasm

Reply:
if I understand your q